src/compas_view2/objects/vectorgroupobject.py

```diff
@@ -0,0 +1,44 @@
+from .object import Object
+from ..buffers import make_index_buffer, make_vertex_buffer
+from compas.utilities import flatten
+
+
+class VectorGroupObject(Object):
+    """Object for displaying vector as arrow sprite."""
+
+    def __init__(self, data, colors=None, color=None, sizes=None, size=None, **kwargs):
+        super().__init__(data, **kwargs)
+        color = color or [0, 0, 0]
+        self.colors = colors or [list(color) for _ in range(len(data.vectors))]
+        size = size or 100
```

Comment:
things like this make it difficult for the user to understand the defaults

Reply:
also not a big fan of differentiating between `size` and `sizes` (or `facecolor` and `facecolors` etc). through a property and corresponding setter we can make sure that whatever value is always converted to a dict, and this dict is always used to assign colors...

Reply:
so i would only keep `size`. if it receives a single value, this is converted to a dict by the property setter. if it receives a dict or a partial dict, they get updated by the setter as well...

Reply:
@brgcode  Do you mean a list ? Perhaps like this?
```python
@property.setter
def size(self, size):
    if isinstance(size, int):
        self._size = [size]
    elif isinstance(size, list):
        self._size = size
    else:
        raise TypeError(".....")
```
Otherwise what should this dict look like?